docs/csharp/misc/cs0250.md

```diff
@@ -10,7 +10,7 @@ ms.assetid: a994f361-6287-4db0-9ce1-e293a8190049
 ---
 # Compiler Error CS0250
 
-Do not directly call your base class Finalize method. It is called automatically from your destructor.  
+Do not directly call your base class Finalize method. It is called automatically from your finalizer.  
```

Comment:
It's correct as it is, as it's produced by the compiler:
![image](https://user-images.githubusercontent.com/15279990/133135499-22a39055-cce8-472e-9b79-4926a6745a7c.png)

Reply:
@pkulikov, so statements at [Finalizers (C# Programming Guide)](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/destructors) does not have precedence over, when issue is (tightly?) compiler related? 

I see [C# 6.0 draft specification](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/classes#destructors) uses _Destructor_ as naming.

Reply:
@pkulikov @Uzivatel919 

These guidelines apply to all these related issues and PRs:

- The [C# 6 draft standard](https://github.com/dotnet/csharpstandard/blob/draft-v6/standard/classes.md#1513-finalizers) uses *finalizers*. That is the correct name going forward. The ECMA committee is currently reconciling the C# 5 standard language with the descriptions of new features added in C# 6.0 in the spec in csharplang.  dotnet/csharpstandard#366 describes the plan of record for the committee and the C# LDM to reconcile these two versions. 
- We should not change the compile warning / error message text in these articles. That language was chosen for search engine optimization. If you search for the error code (for example "CS0260" or the full error message "Do not directly call your base type Finalize method", you'll get these articles high in the search results. The search ranking is more important than the terminology here.
- In other instances, replacing "destructor" with "finalizer" is correct.
-

Reply:
@BillWagner thank you for clarification.
@Uzivatel919 in the articles about compiler errors/warnings, the very first paragraph is the message produced by the compiler; so it shouldn't be changed unless the compiler itself changes the text.

Reply:
Ok. 
1) If this is so, does make any sense to make these terms intermixed on error/warning page?
2) FMI, compiler changes texts by itself?

Reply:
> If this is so, does make any sense to make these terms intermixed on error/warning page?

Yes. That will help readers understand how the platform is evolving.

>  I guess these are mirrored from compiler implementation.

Yes. I don't believe the compiler team is planning any changes here.